tarsier/tag_utils.ts

```diff
@@ -463,21 +463,46 @@ function insertTags(
     }
 
     trimTextNodeStart(element);
-
-    // Base case
     return element;
   }
 
-  const idToTagMetadata: { [key: number]: TagMetadata } = {};
+  function extractIdSymbol(idSpanContent: string): string {
+    const symbols = ["@", "#", "%", "$"];
+    for (const symbol of symbols) {
+      if (idSpanContent.includes(symbol)) {
+        return symbol;
+      }
+    }
+    return "";
+  }
```

Comment:
Manually doing string extraction is unnecessary to do. We should just construct the symbol and pass it down instead. Also we can more strictly enforce expected types this way

```
// Create the standalone function to get the symbol directly
function getTagSymbol(el: HTMLElement): '#' | '$' | '@' | '' {
  if (isInteractable(el)) {
    if (isTextInsertable(el)) return '#';
    else if (el.tagName.toLowerCase() == "a") return'@';
    else return '$';
  } else {
    return '';
  }
}

function create_tagged_span(idNum: number, symbol: string, el: HTMLElement) {
  let idStr = `[${symbol}${idNum}]`;
...
```

``` 
// Now using the create_tagged_span function looks like this
    const symbol = getTagSymbol(el);
    const idSpan = create_tagged_span(idNum, symbol, el);
```

Reply:
Alternatively, you should add it as a data attribute within the span itself to very easily retrieve it